roofit/roofitcore/inc/TestStatistics/kahan_sum.h

```diff
@@ -0,0 +1,88 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+// --- kahan summation templates ---
+
+#ifndef ROOT_ROOFIT_kahan_sum
+#define ROOT_ROOFIT_kahan_sum
+
+#include <map>
+
+namespace RooFit {
+
+template <typename C>
```

Comment:
Suggest replacing this by [`ROOT::Math::KahanSum`](https://root.cern.ch/doc/master/classROOT_1_1Math_1_1KahanSum.html)

Note also that you can go faster if you use that one with multiple accumulators and auto-vectorisation. Users needs to compile themselves with `-march=native` or similar, but it's an easy thing to get more speed.